core/internal/filetransfer/tasks.go

```diff
@@ -32,9 +32,15 @@ type Task struct {
 	// Headers to send on the upload
 	Headers []string
 
-	// Size is the size of the file
+	// Size is the number of bytes to upload
+	//
+	// If this is zero, the entire file is uploaded; if non-zero,
+	// then that many bytes starting from `Offset` are uploaded.
```

Comment:
this isn't accurate to the code---right now, the entire file is uploaded if both Size and Offset are zero, and with my suggestion, Size==0 uploads the rest of the file from Offset